src/databricks/labs/ucx/source_code/notebooks/sources.py

```diff
@@ -61,13 +60,18 @@ def to_migrated_code(self):
             sources.append('')  # following join will append lf
         return '\n'.join(sources)
 
-    def build_dependency_graph(self, parent: DependencyGraph, path_lookup: PathLookup) -> None:
-        path_lookup.push_cwd(self.path.parent)
+    def build_dependency_graph(self, parent: DependencyGraph) -> list[DependencyProblem]:
         problems: list[DependencyProblem] = []
         for cell in self._cells:
-            cell.build_dependency_graph(parent, problems.append)
-        parent.add_problems(problems)
-        path_lookup.pop_cwd()
+            maybe = cell.build_dependency_graph(parent)
```

Comment:
this is where things can go south with the cwd. Since each cell may trigger loading of a child that changes the cwd, not sure how the next cell still uses the cwd of the enclosing Notebook...

Reply:
cwd cannot be changed, per se